src/Strategy/Python/Poetry/PyProject.hs

```diff
@@ -113,10 +115,38 @@ data PyProjectPoetry = PyProjectPoetry
   , version :: Maybe Text
   , description :: Maybe Text
   , dependencies :: Map Text PoetryDependency
-  , devDependencies :: Map Text PoetryDependency
+  , -- For Poetry pre-1.2.x style, understood by Poetry 1.0–1.2
+    devDependencies :: Map Text PoetryDependency
+  , -- Since v1.2.0 of poetry, dependency groups are recommanded way to
+    -- provide development, test, and other optional dependencies.
+    -- refer to: https://python-poetry.org/docs/managing-dependencies#dependency-groups
+    --
+    -- Due to current toml-parsing limitations, we explicitly specify dev, and
+    -- test group only. Note that any dependencies from these groups are excluded
+    -- by default. refer to: https://github.com/kowainik/tomland/issues/336
+    groupDevDependencies :: Map Text PoetryDependency
+  , groupTestDependencies :: Map Text PoetryDependency
   }
   deriving (Show, Eq, Ord)
 
+allPoetryProductionDeps :: PyProject -> Map Text PoetryDependency
+allPoetryProductionDeps project = case pyprojectPoetry project of
+  Just (PyProjectPoetry{dependencies}) -> dependencies
```

Comment:
Not important for this ticket, but in the future we should probably make the `PyProject` type be `PyProject a` where tool specific things live inside the `a`. that way we can specify that a given set of code is only valid for `PyProject Poetry` rather than relying on runtime checks (checking if a field is `Just` or `Nothing` ) like we do here.